test/Image3.rb

```diff
@@ -63,6 +63,17 @@ def setup
     @p = Magick::Image.read(IMAGE_WITH_PROFILE).first.color_profile
   end
 
+  def test_posterize
+    assert_nothing_raised do
+      res = @img.posterize
+      assert_instance_of(Magick::Image, res)
+      assert_not_same(@img, res)
+    end
+    assert_nothing_raised { @img.posterize(5) }
+    assert_nothing_raised { @img.posterize(5, true) }
+    assert_raise(ArgumentError) { @img.posterize(5, true, 'x') }
+  end
```

Comment:
@dlemstra I think we could probably increase the RuboCop threshold for class length for now and leave this where it was. That way we won't have irrelevant changes in the commit once this is merged.

Reply:
Would it be possible to completely ignore the class length in the unit tests? Might be easier to have all the tests in that file?

Reply:
@dlemstra yeah, I think we can do that for now. It would be nice to find a good way to split up the test files at some point, but for now we can just remove the threshold.

Reply:
Might be better to change the limit in a separate PR and changd this PR. I like the structure in the folder with the rspec files. Would you mind explaining me where to change the limit or do the PR yoursrlf?

Reply:
Donezo: https://github.com/rmagick/rmagick/pull/546. Yeah, when we're dealing with classes that have lots of complicated methods, I like structuring test files as 1 file per method. It gives us a lot more flexibility in terms of setting things up specific to the method, and also makes it easier to find exactly what you need when trying to update tests.

Reply:
@mockdeep Updated this PR and removed the commit that moves the unit test. Can you or @Watson1978 take another look at it?